crates/telio-model/src/api_config.rs

```diff
@@ -268,6 +268,13 @@ where
     Ok(Some(eps))
 }
 
+/// Turns on the no link detection mechanism
+#[derive(Default, Copy, Clone, Debug, PartialEq, Eq, Deserialize)]
+pub struct FeatureNoLinkDetection {
+    /// Configurable rtt in seconds
+    pub rtt_seconds: Option<u64>,
```

Comment:
So if the `FeatureNoLinkDetection` is `None` in the main config then it's disabled. What does it mean when the feature is `Some()` but the `rtt_seconds` is `None`? Does it make sense for it to be `None`?
If you want to provide default value, use `#[serde(default = "function")`

Reply:
There are 3 cases:
- no_link_detection is None which means feature is Disabled
- no_link_detection is Some with rtt None which means feature is Enabled with default rtt value
- no_link_detection is Some with rtt Some which means feature is Enabled with custom rtt value

I did not used `#[serde(default = "function")]` because I placed this logic in the From trait implementation of NoLinkDetection config struct from telio-wg. If I had to do that I found useless defaulting rtt here.

```
/// No link detection mechanism config
#[derive(Clone)]
pub enum NoLinkDetection {
    /// No link detection mechanism is disabled
    Disabled,
    /// No link detection mechanism is enabled
    /// Which means that events with link_state will be emitted
    Enabled {
        /// Configurable rtt
        rtt: Duration,
    },
}

impl From<Option<FeatureNoLinkDetection>> for NoLinkDetection {
    fn from(value: Option<FeatureNoLinkDetection>) -> Self {
        match value {
            None => NoLinkDetection::Disabled,
            Some(FeatureNoLinkDetection { rtt_seconds: None }) => NoLinkDetection::Enabled {
                rtt: DEFAULT_CONFIGURABLE_RTT,
            },
            Some(FeatureNoLinkDetection {
                rtt_seconds: Some(rtt),
            }) => NoLinkDetection::Enabled {
                rtt: Duration::from_secs(rtt),
            },
        }
    }
}
```
What do you think?

Reply:
I prefer the default value in the config itself, but I'm not pressing this one. If you think it is better as it is, then let's leave it